internals/api/credential.go

```diff
@@ -25,14 +25,23 @@ var (
 	ErrInvalidRoleARN        = errAPI.Code("invalid_role_arn").StatusError("provided role is not a valid ARN", http.StatusBadRequest)
 )
 
+type CredentialMetadataKey string
+
+// Credential metadata keys
+const (
+	CredentialMetadataAWSKMSKey CredentialMetadataKey = "aws_kms_key"
+	CredentialMetadataAWSRole   CredentialMetadataKey = "aws_role"
+)
+
 // Credential is used to authenticate to the API and to encrypt the account key.
 type Credential struct {
-	AccountID   *uuid.UUID     `json:"account_id"`
-	Type        CredentialType `json:"algorithm"`
-	CreatedAt   time.Time      `json:"created_at"`
-	Fingerprint string         `json:"fingerprint"`
-	Name        string         `json:"name"`
-	Verifier    []byte         `json:"verifier"`
+	AccountID   *uuid.UUID        `json:"account_id"`
+	Type        CredentialType    `json:"algorithm"`
+	CreatedAt   time.Time         `json:"created_at"`
+	Fingerprint string            `json:"fingerprint"`
+	Name        string            `json:"name"`
+	Verifier    []byte            `json:"verifier"`
+	Metadata    map[string]string `json:"metadata,omitempty"`
```

Comment:
`map[string]string` or `map[CredentialMetadataKey]string`?

Reply:
Gave it a second thought. It depends on whether we want to allow arbitrary metadata to be supplied.

We could only require expected metadata to be present (e.g. role and kms key for aws credentials), but allow for more metadata to be supplied.

Reply:
Discussed this in person. We want to restrict the allowed metadata to only the role and kms key and only when the credential type is aws-sts.